File: Python/climbStairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        ctr = 0
        fib = 1
        prev = 0
        cur = 1

        while ctr != n:
            cur = fib
            fib = prev + fib
            prev = cur
            ctr += 1

        return fib

File: JavaScript/climbStairs.js
var climbStairs = function(n) {
  
    let last_num = 0, n_steps = 1;

    for (let i=0; i<n; i++) {
        n_steps += last_num;
        last_num = n_steps - last_num;
    }

    return n_steps;
    
};

console.log(climbStairs(5));